src/js/relayclient/ServerHelper.js

```diff
@@ -1,29 +1,40 @@
 const BN = require('web3').utils.toBN
 
+const { event2topic } = require('./utils')
 // relays are "down-scored" in case they timed out a request.
 // they are "forgiven" after this timeout.
 const DEFAULT_RELAY_TIMEOUT_GRACE_SEC = 60 * 30
 
+// assume veri high stake values for preferred relays, so we don't need
+// to change the filter logic to always support them.
+const PREFERRED_STAKE = 1e30
+const PREFERRED_DELAY = 1e10
 class ActiveRelayPinger {
   // TODO: 'httpSend' should be on a network layer
-  constructor (filteredRelays, httpSend, gasPrice, verbose) {
+  constructor (filteredRelays, httpSend, gasPrice, verbose, nextPinger) {
     this.remainingRelays = filteredRelays.slice()
     this.httpSend = httpSend
     this.pingedRelays = 0
     this.relaysCount = filteredRelays.length
     this.gasPrice = gasPrice
     this.verbose = verbose
+    this.nextPinger = nextPinger
   }
 
   /**
    * Ping those relays that were not returned yet. Remove the returned relay (first to respond) from {@link remainingRelays}
    * @returns the first relay to respond to a ping message. Note: will never return the same relay twice.
    */
   async nextRelay () {
-    if (this.remainingRelays.length === 0) {
-      return null
+    let ret = await this.nextRelay1()
+    // only if no relay found in this ActivePinger, start using the next ActivePinger
+    if (!ret && this.nextPinger) {
+      ret = await this.nextPinger.nextRelay()
     }
+    return ret
+  }
 
+  async nextRelay1 () {
```

Comment:
well, 'nextRelay1' is not a legit name for a function

Reply:
yes, it should be `nextRelayInternal`, so that the logic of chaining a pinger can be outside this method (above)

Reply:
Well, it is still `nextRelay1` after the last commit